### DISCUSSION TITLE
Removal of TestInvoker.GetTaskFromResult

### DISCUSSION THREAD

**Discussion by jaredpar** [xunit/xunit#3026] [Question] [answered]
The context for this question is the following commit 

https://github.com/xunit/xunit/commit/9ebc10c86170e6dc40a41cfcf05ee23a9fc260cd

This commit removed the API `TestInvoker<T>.GetTaskFromResult` from xunit.execution.desktop.dll. That is an API that part of Visual Studio test extensibility was depending on. That is causing me `MissingMethodExceptions` when I move forward to versions of xunit which removed this API.

I need to reach out to the team that owns this to get a fix. But I wanted to understand why this was removed before I reached out to them. Was it wrong for them to depend on this API even though it was public? Is there an alternative that they should be using instead?

**Comment by bradwilson** [comment, marked as answer]
It was moved to a new type and name that better reflected its purpose.

I could reinstate the API (hidden from Intellisense) and forward it to AsyncUtility.TryConvertToTask if needed.

**Reply by jaredpar** [reply]
The place it's getting used is here 

https://github.com/microsoft/vs-extension-testing/blob/main/src/Microsoft.VisualStudio.Extensibility.Testing.Xunit.Shared/Threading/InProcessIdeTestInvoker.cs#L168C13-L168C66

Is it just a 1:1 replacement to move to `AsyncUtility.TryConvertToTask`? If so let me wait to hear from that repo owner on how hard of a change that would be. They're pretty behind in xunit versions so not sure what the cost of them getting to the new API is.

**Reply by bradwilson** [reply]
Restored in https://github.com/xunit/xunit/commit/d97c7f863d8caa88813f940786b97b7c98ae4f52, available in `2.9.1-pre.12` https://xunit.net/docs/using-ci-builds

If needed, I can push this as 2.9.1 ASAP.

**Reply by bradwilson** [reply]
Yes, it's a straight up replacement. I moved it because TestInvoker<T> was a weird place for it to live. I didn't realize anybody else was using it. 😂

**Reply by jaredpar** [reply]
Sorry. I got caught in a series of fire drills last few days. :( 

I was able to work around this by moving back to a version that had the method. That got us past the immediate problem we were having. We can stay there for a bit so there isn't a huge urgency to push this out. 

Very much appreciate your help here. :)
